File: features/promo/tokopoints/src/main/java/com/tokopedia/tokopoints/view/adapter/StackedCouponFilterPagerAdapter.java
package com.tokopedia.tokopoints.view.adapter;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.util.SparseArray;
import android.view.ViewGroup;

import com.tokopedia.tokopoints.view.fragment.CouponListingStackedFragment;
import com.tokopedia.tokopoints.view.fragment.MyCouponListingFragment;
import com.tokopedia.tokopoints.view.model.CouponFilterItem;

import java.util.List;

public class StackedCouponFilterPagerAdapter extends FragmentStatePagerAdapter {

    private List<CouponFilterItem> mItems;
    private SparseArray<Fragment> mrRegisteredFragments = new SparseArray<>();

    public StackedCouponFilterPagerAdapter(FragmentManager fm, List<CouponFilterItem> items) {
        super(fm);
        this.mItems = items;
    }

    @Override
    public Fragment getItem(int position) {
        return CouponListingStackedFragment.newInstance();
    }

    @Override
    public int getCount() {
        return mItems.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mItems.get(position).getName();
    }

    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        Fragment fragment = (Fragment) super.instantiateItem(container, position);
        mrRegisteredFragments.put(position, fragment);
        return fragment;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        mrRegisteredFragments.remove(position);
        super.destroyItem(container, position, object);
    }

    public Fragment getRegisteredFragment(int position) {
        return mrRegisteredFragments.get(position);
    }
}

File: features/promo/tokopoints/src/main/java/com/tokopedia/tokopoints/view/couponlisting/StackedCouponFilterPagerAdapter.kt
package com.tokopedia.tokopoints.view.couponlisting

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import android.util.SparseArray
import android.view.ViewGroup

import com.tokopedia.tokopoints.view.model.CouponFilterItem

class StackedCouponFilterPagerAdapter(fm: FragmentManager, private val mItems: List<CouponFilterItem>) : FragmentStatePagerAdapter(fm) {
    private val mrRegisteredFragments = SparseArray<Fragment>()

    override fun getItem(position: Int): Fragment {
        return CouponListingStackedFragment.newInstance()
    }

    override fun getCount(): Int {
        return mItems.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return mItems[position].name
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val fragment = super.instantiateItem(container, position) as Fragment
        mrRegisteredFragments.put(position, fragment)
        return fragment
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        mrRegisteredFragments.remove(position)
        super.destroyItem(container, position, `object`)
    }

    fun getRegisteredFragment(position: Int): Fragment? {
        return mrRegisteredFragments.get(position)
    }
}
